visualization/app/codeCharta/state/store/metricData/edgeMetricData/edgeMetricData.reducer.ts

```diff
@@ -0,0 +1,118 @@
+	getVisibleFileStates(fileStates).forEach(fileState => {
+		fileState.file.settings.fileSettings.edges.forEach(edge => {
+			if (bothNodesAssociatedAreVisible(edge, allFilePaths, blacklist)) {
+				addEdgeToCalculationMap(edge)
+			}
+		})
+	})
+	return nodeEdgeMetricsMap
+}
+
+function bothNodesAssociatedAreVisible(edge: Edge, filePaths: string[], blacklist: BlacklistItem[]): boolean {
+	const fromPath = filePaths.find(x => x === edge.fromNodeName)
+	const toPath = filePaths.find(x => x === edge.toNodeName)
+	return fromPath && toPath && isNotBlacklisted(fromPath, blacklist) && isNotBlacklisted(toPath, blacklist)
+}
+
+function isNotBlacklisted(path: string, blacklist: BlacklistItem[]): boolean {
+	return !CodeMapHelper.isPathBlacklisted(path, blacklist, BlacklistType.exclude)
+}
+
+function addEdgeToCalculationMap(edge: Edge) {
+	for (const edgeMetric of Object.keys(edge.attributes)) {
+		const edgeMetricEntry = getEntryForMetric(edgeMetric)
+		addEdgeToNodes(edgeMetricEntry, edge.fromNodeName, edge.toNodeName)
+	}
+}
+
+function getEntryForMetric(edgeMetricName: string): Map<string, EdgeMetricCount> {
+	if (!nodeEdgeMetricsMap.has(edgeMetricName)) {
+		nodeEdgeMetricsMap.set(edgeMetricName, new Map())
+	}
+	return nodeEdgeMetricsMap.get(edgeMetricName)
+}
+
+function addEdgeToNodes(edgeMetricEntry: Map<string, EdgeMetricCount>, fromNode: string, toNode: string) {
+	createEntryIfNecessary(edgeMetricEntry, fromNode)
+	createEntryIfNecessary(edgeMetricEntry, toNode)
+	edgeMetricEntry.get(fromNode).outgoing += 1
+	edgeMetricEntry.get(toNode).incoming += 1
+}
+
+function createEntryIfNecessary(edgeMetricEntry: Map<string, EdgeMetricCount>, nodeName: string) {
+	if (!edgeMetricEntry.has(nodeName)) {
+		edgeMetricEntry.set(nodeName, { incoming: 0, outgoing: 0 })
+	}
+}
+
+function getMetricDataFromMap(hashMap: Map<string, Map<string, EdgeMetricCount>>): EdgeMetricData[] {
+	const metricData: EdgeMetricData[] = []
+
+	hashMap.forEach((occurences: any, edgeMetric: any) => {
+		let maximumMetricValue = 0
+		occurences.forEach((value: EdgeMetricCount) => {
+			const combinedValue = value.incoming + value.outgoing
+			if (combinedValue > maximumMetricValue) {
+				maximumMetricValue = combinedValue
+			}
+		})
+		metricData.push({ name: edgeMetric, maxValue: maximumMetricValue })
+	})
+
+	return metricData
+}
+
+function sortNodeEdgeMetricsMap() {
+	const sortedEdgeMetricMap = new Map()
+	if (nodeEdgeMetricsMap) {
+		nodeEdgeMetricsMap.forEach((value, key) => {
+			const sortedMapForMetric = new Map(
+				[...value.entries()].sort((a, b) => b[1].incoming + b[1].outgoing - (a[1].incoming + a[1].outgoing))
+			)
+			sortedEdgeMetricMap.set(key, sortedMapForMetric)
+		})
+	}
+	nodeEdgeMetricsMap = sortedEdgeMetricMap
+}
+
+function addNoneMetric(metricData: EdgeMetricData[]) {
+	if (!metricData.some(x => x.name === EdgeMetricDataService.NONE_METRIC)) {
+		metricData.push({ name: EdgeMetricDataService.NONE_METRIC, maxValue: 0 })
```

Comment:
This could be done in `getMetricDataFromMap`. That way there's no need to go over all entries again.